depresolve.go

```diff
@@ -214,3 +210,24 @@ func standardRepoHostImportPathToCloneURL(importPath string) (string, error) {
 	}
 	return "https://" + strings.Join(parts[:3], "/") + ".git", nil
 }
+
+// replaceImportPathRepoRoot modifies the given importPath by replacing the
+// root string in the importPath with the host+path of the clone URL.
+// This is necessary for resolving refs to custom import path packages, since
+// the defs within those packages would have the Unit field set to
+// "${repoURI}/path/to/pkg/dir", where repoURI is the host+path of the cloneURL.
+//
+// This is a HACK to make def resolution work in presence of custom import
+// paths. A proper solution would require passing in the custom import path
+// information from srclib to the graph step, and setting that as the Unit field
+// of the defs identified in the source unit.
+//
+// TODO: Implement the proper solution and get rid of this hack.
+func replaceImportPathRepoRoot(importPath, root, cloneURL string) string {
+	i := strings.Index(cloneURL, "://")
+	if i < 0 {
+		return importPath
+	}
+	newRoot := strings.TrimSuffix(cloneURL[i+len("://"):], ".git")
```

Comment:
Repo cloning is not handled by `srclib-go` (or `srclib`). The repo is cloned to local disk (usually in the drone build's clone step in Sourcegraph frontend) and `srclib` is called on the local directory. I don't think the `newRoot` here really matters for cloning the repo. Here a "canonical import path" is being constructed by concatenating the repo URI and the package dir path. AFAIK we don't keep the `.git` suffix in repo URIs, so this is consistent with rest of the codebase.